cmd/server/assets/codes/bulkissue.html

```diff
@@ -0,0 +1,289 @@
+          let batch = [];
+          totalUsersCreated = 0;
+          $tableBody.empty();
+          for (let i = parseInt($startAt.val()); i < rows.length && !cancelUpload; i++) {
+            if (rows[i].trim() != "") {
+              if (batch.length >= batchSize) {
+                $tableBody.empty();
+                batch = [];
+              }
+
+              let request = {};
+              let cols = rows[i].split(',');
+              request["phone"] = cols[0].trim();
+              request["symptomDate"] = (cols.length > 1) ? cols[1].trim() : "";
+              let hs  = String(CryptoJS.HmacSHA256(request["phone"], retryCode)).substr(0,36)
+              request["uuid"] = hs.substr(0,8)+ '-' + hs.substr(9,4) + '-' + hs.substr(13,4) + '-' + hs.substr(17,4) + '-' + hs.substr(21,12);
+
+              let row = "<tr><td>" + request["phone"] + "</td><td>" + request["symptomDate"] + "</td></tr>";
+              $tableBody.append(row);
+
+              batch.push(request);
+            }
+
+            if (batch.length >= batchSize || i == rows.length - 1) {
+              await uploadBatch(batch).catch(err => { });
+              $startAt.val(i);
+              if (cancelUpload) {
+                flash.warning("Successfully issued " + total + " codes."
+                  + (rows.length - i) + " remaining.");
+                break;
+              }
+
+              let percent = Math.floor((i + 1) * 100 / rows.length) + "%";
+              $progress.width(percent);
+              $progress.html(percent);
+            }
+          }
+
+          if (!cancelUpload) {
+            flash.alert("Successfully issued" + total + " codes.");
+          }
+          $table.fadeOut(400);
+          $import.prop('disabled', false);
+          $cancel.prop('disabled', true);
+        };
+
+        return { start, cancel };
+      }
+    });
+
+    function uploadBatch(data) {
+      return $.ajax({
+        type: 'POST',
+        url: '/code/issue',
+        // TODO: at the moment we only support one-at-a-time
+        data: JSON.stringify(data[0]),
+        headers: { 'X-CSRF-Token': '{{.csrfToken}}' },
+        contentType: 'application/json',
+        success: function(result) {
+          total += result.newUsers.length;
+          if (result.error) {
+            flash.error(result.error, "danger");
+          }
+        },
+        error: function(xhr, status, e) {
+          flash.error(e, "danger");
+        },
+      });
+    }
+
+    // generates a random 16 digit alphanumeric code
+    function genCode() {
+      return Math.random().toString(36).substr(2, 8) + Math.random().toString(36).substr(2, 8);
+    }
+
+    function setCookie(cname, cvalue, exdays) {
+      var d = new Date();
+      d.setTime(d.getTime() + (exdays * 24 * 60 * 60 * 1000));
+      var expires = "expires="+d.toUTCString();
+      document.cookie = cname + "=" + cvalue + ";" + expires;
+    }
+
+    function getCookie(cname) {
```

Comment:
We should move cookie functions into `application.js`. We should probably move all of this into application.js

Reply:
moved